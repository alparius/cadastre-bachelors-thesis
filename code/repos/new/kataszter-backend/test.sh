#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
