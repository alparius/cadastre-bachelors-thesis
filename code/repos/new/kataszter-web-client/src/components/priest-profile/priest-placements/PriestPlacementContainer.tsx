import React from "react";
import { Button, Divider, Header, Icon } from "semantic-ui-react";

import IParishMin from "../../../data/parish/ParishMin";
import { NewMainPlacement } from "../../../data/priest/MainPlacement";
import { NewPlacement } from "../../../data/priest/Placement";
import IPriestProfileData from "../../../data/priest/PriestProfileData";
import useFetchData from "../../../requests/useFetchData";
import BoldFM from "../../../util/styledMessages/BoldFM";
import ItalicFM from "../../../util/styledMessages/ItalicFM";
import ErrorMsg from "../../ErrorMsg";
import ProfilePlaceholder from "../../ProfilePlaceholder";
import PriestAssistantPriestPlaceList from "./PriestAssistantPriestPlaceList";
import PriestMainPriestPlaceList from "./PriestMainPriestPlaceList";

interface IPriestPlacementContainerProps {
    priest: IPriestProfileData;
    loading: boolean;
    showError: boolean;
    editMode: boolean;
    setPriest(newPriest: IPriestProfileData): void;
}

const PriestPlacementContainer: React.FC<IPriestPlacementContainerProps> = (props: IPriestPlacementContainerProps) => {
    const { priest, loading, showError, editMode, setPriest } = props;

    const { response: parishResponse, loading: parishLoading, showError: parishShowError } = useFetchData<IParishMin[]>(`/parishes`);

    const handleNewMainPlacement = () => {
        if (priest) {
            if (priest.mainPriestPlaces) {
                priest.mainPriestPlaces.splice(0, 0, { ...NewMainPlacement });
                setPriest({
                    ...priest,
                    mainPriestPlaces: priest.mainPriestPlaces
                });
            } else {
                setPriest({
                    ...priest,
                    mainPriestPlaces: [{ ...NewMainPlacement }]
                });
            }
        }
    };

    const handleNewAssistantPlacement = () => {
        if (priest) {
            if (priest.assistantPriestPlaces) {
                priest.assistantPriestPlaces.splice(0, 0, { ...NewPlacement });
                setPriest({
                    ...priest,
                    assistantPriestPlaces: priest.assistantPriestPlaces
                });
            } else {
                setPriest({
                    ...priest,
                    assistantPriestPlaces: [{ ...NewPlacement }]
                });
            }
        }
    };

    if (showError) {
        return <ErrorMsg errorMessage={"error.connection"} />;
    } else if (loading || !priest) {
        return <ProfilePlaceholder />;
    } else {
        return (
            <>
                <Header as="h2">
                    {priest.mainPriestPlaces && priest.mainPriestPlaces.length > 1 ? (
                        <ItalicFM id="profile.mainPriestPlaces" />
                    ) : (
                        <ItalicFM id="profile.mainPriestPlace" />
                    )}
                </Header>
                {editMode && (
                    <Button
                        id="addNewMainPlacement"
                        style={{ marginBottom: "2em" }}
                        icon
                        basic
                        color="violet"
                        circular
                        labelPosition="left"
                        onClick={handleNewMainPlacement}
                    >
                        <Icon name="add" color="violet" size="large" />
                        <BoldFM id="profile.addNewMainPlacement" />
                    </Button>
                )}
                <PriestMainPriestPlaceList
                    priest={priest}
                    editMode={editMode}
                    setPriest={setPriest}
                    parishResponse={parishResponse}
                    parishLoading={parishLoading}
                    parishShowError={parishShowError}
                />

                <Divider style={{ marginTop: "2em" }} />

                <Header as="h2">
                    {priest.assistantPriestPlaces && priest.assistantPriestPlaces.length > 1 ? (
                        <ItalicFM id="profile.assistantPriestPlaces" />
                    ) : (
                        <ItalicFM id="profile.assistantPriestPlace" />
                    )}
                </Header>
                {editMode && (
                    <Button
                        id="addNewAssistantPlacement"
                        icon
                        basic
                        color="violet"
                        circular
                        labelPosition="left"
                        onClick={handleNewAssistantPlacement}
                    >
                        <Icon name="add" color="violet" size="large" />
                        <BoldFM id="profile.addNewAssistantPlacement" />
                    </Button>
                )}
                <PriestAssistantPriestPlaceList
                    priest={priest}
                    editMode={editMode}
                    setPriest={setPriest}
                    parishResponse={parishResponse}
                    parishLoading={parishLoading}
                    parishShowError={parishShowError}
                />
            </>
        );
    }
};

export default PriestPlacementContainer;
