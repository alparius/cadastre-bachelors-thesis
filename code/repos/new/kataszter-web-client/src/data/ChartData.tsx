interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
