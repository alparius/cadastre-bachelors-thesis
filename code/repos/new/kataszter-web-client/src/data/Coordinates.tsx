interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
