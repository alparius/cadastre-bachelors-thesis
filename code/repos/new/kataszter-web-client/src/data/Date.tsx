interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
