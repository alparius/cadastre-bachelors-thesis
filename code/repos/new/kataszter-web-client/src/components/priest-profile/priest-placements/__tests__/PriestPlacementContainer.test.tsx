/*===================
  PriestPlacementContainer.unit.test.tsx
 ===================*/
import React from "react";

import IPriestProfileData from "../../../../data/priest/PriestProfileData";
import { NewTestPriestProfileData } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntlAndRouter } from "../../../../util/test/test-wrapper";
import PriestPlacementContainer from "../PriestPlacementContainer";

const handlers = {
    setPriest: jest.fn(),
    setEditMode: jest.fn()
};

const priest: IPriestProfileData = NewTestPriestProfileData;

describe("PriestPlacementContainer element", () => {
    afterEach(() => {
        handlers.setPriest.mockClear();
        handlers.setEditMode.mockClear();
    });

    it("should render a placement container when the priest is not empty", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestPlacementContainer priest={priest} loading={false} showError={false} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder when the priest is empty", () => {
        const emptyPriestFamily: IPriestProfileData[] = [];
        const container = createComponentWithIntlAndRouter(
            <PriestPlacementContainer priest={emptyPriestFamily[0]} loading={false} showError={false} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError is true", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestPlacementContainer priest={priest} loading={false} showError editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError and loading are both true", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestPlacementContainer priest={priest} loading showError editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card when only loading is true", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestPlacementContainer priest={priest} loading showError={false} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call setPriest if add new main placement Button pressed", () => {
        const container = mountWithIntlAndRouter(
            <PriestPlacementContainer priest={priest} loading={false} showError={false} editMode {...handlers} />
        );
        container.find("#addNewMainPlacement").at(0).simulate("click");
        expect(handlers.setPriest).toHaveBeenCalledTimes(1);
    });

    it("should call setPriest if add new assistant placement Button pressed", () => {
        const container = mountWithIntlAndRouter(
            <PriestPlacementContainer priest={priest} loading={false} showError={false} editMode {...handlers} />
        );
        container.find("#addNewAssistantPlacement").at(0).simulate("click");
        expect(handlers.setPriest).toHaveBeenCalledTimes(1);
    });
});
