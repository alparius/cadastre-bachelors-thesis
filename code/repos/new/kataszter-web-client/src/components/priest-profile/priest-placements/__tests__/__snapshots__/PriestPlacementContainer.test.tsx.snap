// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestPlacementContainer element should render a placeholder card when only loading is true 1`] = `
Array [
  <div
    className="ui four column grid"
  >
    <div
      className="column"
    />
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
      </div>
    </div>
    <div
      className="column"
    />
  </div>,
  <div
    className="ui divider"
  />,
  <div
    className="ui four column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
  </div>,
]
`;

exports[`PriestPlacementContainer element should render a placeholder when the priest is empty 1`] = `
Array [
  <div
    className="ui four column grid"
  >
    <div
      className="column"
    />
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
      </div>
    </div>
    <div
      className="column"
    />
  </div>,
  <div
    className="ui divider"
  />,
  <div
    className="ui four column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      />
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line medium"
          />
        </div>
      </div>
      <div
        className="column"
      />
    </div>
  </div>,
]
`;

exports[`PriestPlacementContainer element should render a placement container when the priest is not empty 1`] = `
Array [
  <h2
    className="ui header"
  >
    <i>
      Main priest place
    </i>
  </h2>,
  <div
    className="ui doubling stackable two cards"
  >
    <a
      aria-current={null}
      className="ui fluid card"
      href="/parishes/profile/0"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="content"
        >
          <div
            className="header"
            id="header"
          >
            <h2
              className="ui header"
            >
              place
            </h2>
          </div>
        </div>
        <div
          className="description"
        >
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Start date
                    :
                  </b>
                   
                  <span
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    2000.06.06
                  </span>
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    End date
                    :
                  </b>
                   
                  <span
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    2000.06.06
                  </span>
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Number of people
                    :
                  </b>
                   
                  numberOfPeople
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Muniment issuing authority
                    :
                  </b>
                   
                  munimentIssuingAuthority
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Muniment genesis date
                    :
                  </b>
                   
                  <span
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    2000.06.06
                  </span>
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Muniment number
                    :
                  </b>
                   
                  munimentNumber
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Income
                    :
                  </b>
                   
                  income
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>,
  <div
    className="ui divider"
    style={
      Object {
        "marginTop": "2em",
      }
    }
  />,
  <h2
    className="ui header"
  >
    <i>
      Assistant priest place
    </i>
  </h2>,
  <div
    className="ui doubling stackable two cards"
  >
    <a
      aria-current={null}
      className="ui fluid card"
      href="/parishes/profile/0"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="content"
        >
          <div
            className="header"
            id="header"
          >
            <h2
              className="ui header"
            >
              place
            </h2>
          </div>
        </div>
        <div
          className="description"
        >
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    Start date
                    :
                  </b>
                   
                  <span
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    2000.06.06
                  </span>
                </div>
              </form>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline field"
                >
                  <b>
                    End date
                    :
                  </b>
                   
                  <span
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    2000.06.06
                  </span>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>,
]
`;

exports[`PriestPlacementContainer element should render an error message when showError and loading are both true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`PriestPlacementContainer element should render an error message when showError is true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;
